src/language-yaml/printer-yaml.js

```diff
@@ -713,12 +713,13 @@ function printNextEmptyLine(path, originalText) {
   const node = path.getValue();
   const root = path.stack[0];
 
-  root.isNextEmptyLinePrintedChecklist =
-    root.isNextEmptyLinePrintedChecklist || [];
+  if (!root.isNextEmptyLinePrintedSet) {
+    root.isNextEmptyLinePrintedSet = new Set();
+  }
```

Comment:
I looked this part too yesterday, my first thought is `cache(node)`, but turns out, it's not, it check `end.line`.

I think we should not cache this in root, how about


```js
const cache = new WeakMap();
function printNextEmptyLine(path, originalText) {
  // ...

  let isNextEmptyLinePrintedSet
  if (cache.has(root)) {
    isNextEmptyLinePrintedSet = cache.get(root)
  } else {
    isNextEmptyLinePrintedSet = new Set()
    cache.set(root, isNextEmptyLinePrintedSet )
  }

  // ...
}
```